### DISCUSSION TITLE
Controller vs useController

### DISCUSSION THREAD

**Discussion by msageryd** [react-hook-form/react-hook-form#7554] [Q&A] [answered]
Sorry for another newbie question.

I think code with render-functions as props is harder to read, so I prefer `useController` over `<Controller>`. All examples in the docs use Controller, so I thought I should ask if there is any reason for this.

Is there any difference between the two concepts?

**Comment by Moshyfawn** [comment, marked as answer]
`useController` hook is the same as `<Controller />` functionality-wise, just hooks. Additionally, `<Controller />` component is powered by `useController` internally ;)

**Reply by msageryd** [reply]
Thanks.
I just realized that I didn't search before asking. Just found this [#6157](https://github.com/react-hook-form/react-hook-form/discussions/6157)

**Reply by Moshyfawn** [reply]
Fair enough :P
Yep, it's headless UI vs components and the prior feels cleaner. Tho, I might occasionally use `<Controller />` for the sake of quickly prototyping a form if there's a new input component that needs introduction

**Reply by vltansky** [reply]
I created this example to demonstrate the difference. Hope it helps someone :)
https://stackblitz.com/edit/stackblitz-starters-gyqpdm?file=app%2Fpage.tsx
